app/javascript/mastodon/features/ui/components/sign_in_banner.js

```diff
@@ -1,13 +1,35 @@
 import React from 'react';
 import { FormattedMessage } from 'react-intl';
+import { useDispatch } from 'react-redux';
 import { registrationsOpen } from 'mastodon/initial_state';
+import { openModal } from 'mastodon/actions/modal';
 
-const SignInBanner = () => (
-  <div className='sign-in-banner'>
-    <p><FormattedMessage id='sign_in_banner.text' defaultMessage='Sign in to follow profiles or hashtags, favourite, share and reply to posts, or interact from your account on a different server.' /></p>
-    <a href='/auth/sign_in' className='button button--block'><FormattedMessage id='sign_in_banner.sign_in' defaultMessage='Sign in' /></a>
-    <a href={registrationsOpen ? '/auth/sign_up' : 'https://joinmastodon.org/servers'} className='button button--block button-tertiary'><FormattedMessage id='sign_in_banner.create_account' defaultMessage='Create account' /></a>
-  </div>
-);
+const SignInBanner = () => {
+  const dispatch = useDispatch();
+
+  let signupButton;
+
+  if (registrationsOpen) {
+    signupButton = (
+      <a href='/auth/sign_up' className='button button--block button-tertiary'>
+        <FormattedMessage id='sign_in_banner.create_account' defaultMessage='Create account' />
+      </a>
+    );
+  } else {
+    signupButton = (
+      <button className='button button--block button-tertiary' onClick={() => dispatch(openModal('CLOSED_REGISTRATIONS'))}>
```

Comment:
If you want to keep this as a functional component, you might need to use `useRef` to set event handlers.

https://bobbyhadz.com/blog/react-functional-component-add-event-listener

Reply:
The code is sound, it's just that the anonymous function cannot get compared, but I really don't think this matters here. I don't understand your suggestion, seems to me the pattern would be to use `useCallback`, not a clunky `useRef` + `useEffect`.